_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -378,7 +362,7 @@ public static function get_user_connection_data_permission_callback() {
 	 * @return bool|WP_Error True if user is able to unlink.
 	 */
 	public static function unlink_user_permission_callback() {
-		if ( current_user_can( 'jetpack_connect' ) || Jetpack::is_user_connected( get_current_user_id() ) ) {
+		if ( current_user_can( 'jetpack_connect_user' ) && Jetpack::is_user_connected( get_current_user_id() ) ) {
```

Comment:
So, some minor concern here:

https://github.com/Automattic/jetpack/blob/327b2f8979695192cccecc7fd1b4ebfeaf195cc9/class.jetpack.php#L692-L698

Firstly, I feel we should probably split `jetpack_disconnect_user` into a second permission check from the connect one.  It may be desired to only let a user connect themselves, but let other admins disconnect them -- or something like that.

But more concerningly is the fact that the endpoint in question (`::unlink_user`) can take a User ID argument -- but the cap we're checking just maps to whether they have the `read` cap -- shouldn't we have something in place here that will reject the request if one user is trying to disconnect another user, without the proper permissions?

My concern is that a contributor could potentially unlink an admin. :\